pkg/acme/integration.go

```diff
@@ -107,3 +115,60 @@ func (i *Integration) IssueCertificates() (reloadRequired bool, err error) {
 
 	return reloadRequired, err
 }
+
+func (i *Integration) ScheduleRenewals() (reloadRequired bool) {
+	reloadRequired = false
+	const CertificateExpiryThreshold = 720 // 720 hours = 30 days
+
+	if len(i.renewalList) == 0 {
+		i.logger.Debugf("No certificates to watch for renewal")
+		return reloadRequired
+	}
+
+	for primaryDomain := range i.renewalList {
+		domains := i.renewalList[primaryDomain]
+
+		certBytes, keyBytes, err := i.certStorage.GetCertificate(primaryDomain, domains)
+		if err != nil {
+			i.logger.Warnf("skipped renewal check for %s due to storage error", primaryDomain)
+			continue
+		}
+
+		pair, err := tls.X509KeyPair(certBytes, keyBytes)
+		if err != nil {
+			i.logger.Warnf("parsing certificate from storage failed: %", err.Error())
+			continue
+		}
+
+		cert, _ := x509.ParseCertificate(pair.Certificate[0])
+		if time.Now().Add(CertificateExpiryThreshold * time.Hour).After(cert.NotAfter) {
+			go i.addToIssueBacklog(domains)
+			i.logger.Infof("queued renewal of certificate for %s", primaryDomain)
+			reloadRequired = true
+		}
+	}
+
+	return reloadRequired
+}
+
+func (i *Integration) addToRenewalList(domains []string) {
+	backlogKey := domains[0] // @see TestVhostPrimaryDomainIsFirstInDomains
```

Comment:
This assumption is done multiple times in the TLS package. Let's create an improvement to migrate this into an embedded struct